api/uniter/unit_test.go

```diff
@@ -239,6 +239,35 @@ func (s *unitSuite) TestWatch(c *gc.C) {
 	wc.AssertOneChange()
 }
 
+func (s *unitSuite) TestWatchRelations(c *gc.C) {
+	w, err := s.apiUnit.WatchRelations()
+	c.Assert(err, jc.ErrorIsNil)
+	wc := watchertest.NewStringsWatcherC(c, w, s.BackingState.StartSync)
+	defer wc.AssertStops()
+
+	// Initial event.
+	wc.AssertChange()
+	wc.AssertNoChange()
+
+	// Change something other than the lifecycle and make sure it's
+	// not detected.
+	err = s.wordpressService.SetExposed()
+	c.Assert(err, jc.ErrorIsNil)
+	wc.AssertNoChange()
+
+	// Add another service and relate it to wordpress,
```

Comment:
There's one other case that I know of that uses non-trivial non-regexp type filter - WatchRemoteRelations()
It could be argued that that other case should have been done like this, to avoid blocking the watcher queue on db lookups etc. I think this is probably ok, and maybe we should look to port the WatchRemoteRelations() to use this style